spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -244,6 +244,7 @@ private boolean requiresRewrite(Filter filter, Schema schema, Set<Integer> ident
 
   @Override
   public void deleteWhere(Filter[] filters) {
+    canDeleteWhere(filters);
```

Comment:
This is where the check is done for the other versions of Spark, but adding `canDeleteWhere` here is expensive and not the right solution. Can you add the precondition instead?

Reply:
@rdblue Thanks for the review!